app/models/report.rb

```diff
@@ -478,8 +613,217 @@ def self.report_trending_search(report)
+
+    return if mod_ids.empty?
+
+    time_read_query = <<~SQL
+    SELECT SUM(time_read) AS time_read,
+    user_id
+    FROM user_visits
+    WHERE user_id = ANY(ARRAY#{mod_ids})
+    AND visited_at >= '#{report.start_date}'
+    AND visited_at <= '#{report.end_date}'
+    GROUP BY user_id
+    SQL
+
+    flag_count_query = <<~SQL
+    SELECT agreed_by_id AS user_id,
+    COUNT(*) AS flag_count
+    FROM post_actions
+    WHERE (agreed_by_id = ANY(ARRAY#{mod_ids}) OR disagreed_by_id = ANY(ARRAY#{mod_ids}))
+    AND post_action_type_id = ANY(ARRAY#{PostActionType.flag_types_without_custom.values})
+    AND created_at >= '#{report.start_date}'
+    AND created_at <= '#{report.end_date}'
+    GROUP BY agreed_by_id
+    SQL
+
+    topic_count_query = <<~SQL
+    SELECT user_id,
+    COUNT(*) AS topic_count
+    FROM topics
+    WHERE user_id = ANY(ARRAY#{mod_ids})
+    AND created_at >= '#{report.start_date}'
+    AND created_at <= '#{report.end_date}'
+    GROUP BY user_id
+    SQL
+
+    post_count_query = <<~SQL
+    SELECT user_id,
+    COUNT(*) AS post_count
+    FROM posts
+    WHERE user_id = ANY(ARRAY#{mod_ids})
+    AND created_at >= '#{report.start_date}'
+    AND created_at <= '#{report.end_date}'
+    GROUP BY user_id
+    SQL
+
+    DB.query(time_read_query).each do |row|
+      mod_data[row.user_id][:time_read] = row.time_read if row.user_id
+    end
+
+    DB.query(flag_count_query).each do |row|
+      mod_data[row.user_id][:flag_count] = row.flag_count if row.user_id
+    end
+
+    DB.query(topic_count_query).each do |row|
+      mod_data[row.user_id][:topic_count] = row.topic_count if row.user_id
+    end
+
+    DB.query(post_count_query).each do |row|
+      mod_data[row.user_id][:post_count] = row.post_count if row.user_id
+    end
+
+    report.data = mod_data.values
+  end
+
+  def self.report_flags_status(report)
+    report.modes = [:table]
+
+    report.labels = [
+      { properties: [:action_type], title: I18n.t("reports.flags_status.labels.flag") },
+      { type: :link, properties: [:staff_username, :staff_url], title: I18n.t("reports.flags_status.labels.assigned") },
+      { type: :link, properties: [:poster_username, :poster_url], title: I18n.t("reports.flags_status.labels.poster") },
+      { type: :link, properties: [:flagger_username, :flagger_url], title: I18n.t("reports.flags_status.labels.flagger") },
+      { type: :seconds, properties: [:response_time], title: I18n.t("reports.flags_status.labels.time_to_resolution") }
+    ]
+
+    report.data = []
+
+    flag_types = PostActionType.flag_types_without_custom
+
+    sql = <<~SQL
+    SELECT pa.post_action_type_id,
+    p.user_id AS poster_id,
+    pa.post_action_type_id,
+    pa.created_at,
+    pa.agreed_at,
+    pa.disagreed_at,
+    pa.deferred_at,
+    pa.agreed_by_id,
+    pa.disagreed_by_id,
+    pa.deferred_by_id,
+    pa.user_id AS flagger_id,
+    (select u.username FROM users u WHERE u.id = pa.user_id) AS flagger_username,
```

Comment:
I think we should do a join on the users table instead since this will cause PostgreSQL to run a sub-queury for each row returned.